core/src/main/scala/io/projectglow/sql/expressions/FirthTest.scala

```diff
@@ -161,6 +161,7 @@ class FirthNewtonArgs(numRows: Int, numCols: Int) {
 
   def initFromNullFit(nullFit: FirthNewtonArgs): Unit = {
     b(0 until nullFit.b.length) := nullFit.b
+    b(nullFit.b.length to -1) := 0d
```

Comment:
I couldn't reproduce the same failure mode for Firth logistic regression, but we should have the same re-initialization logic here.

Reply:
I think for firth we `clear` before each fit because we are not caching and `clear` takes care of it.

Reply:
Ah, yes